Assets/MixedRealityToolkit/InputModule/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,631 @@
+        public event Action StartedManipulating;
+        // Event that gets raised when the user ends manipulation
+        public event Action StoppedManipulating;
+        // Bounding Box visual on manipulation interaction
+        public GameObject GlobalBoundingBox;
+
+        [Tooltip("Transform that will be dragged. Defaults to the object of the component.")]
+        public Transform HostTransform;
+
+        public enum TwoHandedManipulation
+        {
+            Scale,
+            Rotate,
+            MoveScale,
+            RotateScale,
+            MoveRotateScale
+        };
+
+        [Tooltip("What manipulation will two hands perform?")]
+        public TwoHandedManipulation ManipulationMode;
+
+        [Tooltip("Constrain rotation along an axis")]
+        public HandlebarRotateLogic.RotationConstraint ConstraintOnRotation = HandlebarRotateLogic.RotationConstraint.None;
+        [Tooltip("If true, grabbing the object with one hand will initiate movement.")]
+        public bool OneHandMovement = true;
+
+        [Flags]
+        private enum State
+        {
+            Start = 0x000,
+            Moving = 0x001,
+            Scaling = 0x010,
+            Rotating = 0x100,
+            MovingScaling = 0x011,
+            RotatingScaling = 0x110,
+            MovingRotatingScaling = 0x111
+        };
+
+        private State currentState;
+        private MoveSphericalCoordsLogic m_moveLogic;
+        private ScaleLogic m_scaleLogic;
+        private HandlebarRotateLogic m_rotateLogic;
+        // Maps input id -> position of hand
+        private readonly Dictionary<uint, Vector3> m_handsPressedLocationsMap = new Dictionary<uint, Vector3>();
+        // Maps input id -> input source. Then obtain position of input source using currentInputSource.TryGetGripPosition(currentInputSourceId, out inputPosition);
+        private readonly Dictionary<uint, IInputSource> m_handsPressedInputSourceMap = new Dictionary<uint, IInputSource>();
+
+        private void Awake()
+        {
+            m_moveLogic = new MoveSphericalCoordsLogic();
+            m_rotateLogic = new HandlebarRotateLogic(ConstraintOnRotation);
+            m_scaleLogic = new ScaleLogic();
+
+            GlobalBoundingBox = GameObject.Find("BoundingBoxBasic");
+        }
+
+        private void Update()
+        {
+            // Update positions of all hands
+            foreach (var key in m_handsPressedInputSourceMap.Keys)
+            {
+                var inputSource = m_handsPressedInputSourceMap[key];
+                Vector3 inputPosition = Vector3.zero;
+                if (inputSource.TryGetGripPosition(key, out inputPosition))
+                {
+                    m_handsPressedLocationsMap[key] = inputPosition;
+                }
+            }
+
+            if (currentState != State.Start)
+            {
+                UpdateStateMachine();
+            }
+        }
+
+        private void Start()
```

Comment:
please move the start method between awake and update.

Reply:
Done.